File: src/main/java/com/github/joelws/release/tracker/boot/Application.java
package com.github.joelws.release.tracker.boot;

import org.apache.cxf.jaxrs.JAXRSServerFactoryBean;
import org.springframework.context.support.ClassPathXmlApplicationContext;

public class Application {
    public static void main(String... args) {
        Application application = new Application();
        application.run();
    }

    void run() {
        ClassPathXmlApplicationContext ctx = new ClassPathXmlApplicationContext("META-INF/beans.xml");
        JAXRSServerFactoryBean sf = (JAXRSServerFactoryBean) ctx.getBean("releaseTracker");
        sf.init();
    }
}

File: src/main/kotlin/com/github/joelws/release/tracker/boot/Application.kt
package com.github.joelws.release.tracker.boot

import com.github.joelws.release.tracker.configuration.AppConfig
import org.springframework.boot.SpringApplication
import org.springframework.boot.autoconfigure.EnableAutoConfiguration
import org.springframework.boot.orm.jpa.EntityScan
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.Import
import org.springframework.context.annotation.ImportResource
import org.springframework.transaction.annotation.EnableTransactionManagement

@Configuration
@EntityScan(basePackages = arrayOf("com.github.joelws.release.tracker.entity"))
@Import(AppConfig::class)
@EnableAutoConfiguration(exclude = arrayOf(AppConfig::class))
@EnableTransactionManagement
@ImportResource("classpath:META-INF/cxf/cxf.xml", "classpath:META-INF/cxf/cxf-servlet.xml")
open class Application {

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            SpringApplication.run(Application::class.java, *args)
        }

    }


}


